calyx-py/calyx/builder.py

```diff
@@ -620,13 +617,20 @@ def port(self, name: str) -> ExprBuilder:
         """Build a port access expression."""
         return ExprBuilder(ast.Atom(ast.CompPort(self._cell.id, name)))
 
-    def is_mem_d1(self) -> bool:
+    def is_std_mem_d1(self) -> bool:
         """Check if the cell is a StdMemD1 cell."""
         return (
             isinstance(self._cell.comp, ast.CompInst)
             and self._cell.comp.id == "std_mem_d1"
         )
 
+    def is_seq_mem_d1(self) -> bool:
```

Comment:
Might want to define a `is_primitive(prim_name)` method for this kind of check

Reply:
Nice, done!